server/src/main/java/io/pravega/schemaregistry/rules/jsoncompatibility/ArrayTypeComparator.java

```diff
@@ -0,0 +1,133 @@
+package io.pravega.schemaregistry.rules.jsoncompatibility;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Iterators;
+import io.pravega.schemaregistry.rules.CompatibilityChecker;
+
+import java.util.Iterator;
+
+import static io.pravega.schemaregistry.rules.jsoncompatibility.BreakingChangesStore.BreakingChanges;
+
+public class ArrayTypeComparator {
+    JsonCompatibilityChecker jsonCompatibilityChecker;
+    public void setJsonTypeComparator() {
+        this.jsonCompatibilityChecker = new JsonCompatibilityChecker();
+    }
+    public BreakingChanges compareArrays(JsonNode toCheck, JsonNode toCheckAgainst) {
+        if (toCheck.isArray() && toCheckAgainst.isArray())
+            return arraySimpleBodyComparision(toCheck, toCheckAgainst);
+        else {
+            if (checkUniqueItems(toCheck, toCheckAgainst) != null)
+                return checkUniqueItems(toCheck, toCheckAgainst);
+            BreakingChanges minMaxChanges = minMaxItems(toCheck, toCheckAgainst);
+            if (minMaxChanges != null)
+                return minMaxChanges;
+            BreakingChanges additionalItemsChanges = additionalItems(toCheck, toCheckAgainst);
+            if (additionalItemsChanges != null)
+                return additionalItemsChanges;
+            BreakingChanges itemsValidationChanges = itemValidation(toCheck, toCheckAgainst);
+            if(itemsValidationChanges != null)
+                return itemsValidationChanges;
+            //TO DO: Add contains and tupleValidation
+        }
+        return null;
+    }
+
+    private BreakingChanges arraySimpleBodyComparision(JsonNode toCheck, JsonNode toCheckAgainst) {
+        Iterator<String> allNodes = Iterators.concat(toCheck.fieldNames(), toCheckAgainst.fieldNames());
+        while(allNodes.hasNext()) {
+            String item = allNodes.next();
+            if(!toCheck.has(item))
+                return BreakingChanges.ARRAY_SIMPLE_BODY_CHECK_ELEMENT_REMOVED;
+            else if (!toCheckAgainst.has(item))
+                return BreakingChanges.ARRAY_SIMPLE_BODY_CHECK_ELEMENT_ADDED;
+        }
+        return null;
+    }
+
+    private BreakingChanges checkUniqueItems(JsonNode toCheck, JsonNode toCheckAgainst) {
+        if(toCheck.has("uniqueItems")) {
+            if (toCheck.get("uniqueItems").isBoolean() && toCheck.get("uniqueItems").asText() == "true") {
+                if (toCheckAgainst.get("uniqueItems") == null)
+                    return BreakingChanges.ARRAY_UNIQUE_ITEMS_CONDITION_ENABLED;
```

Comment:
both this and following conditional blocks are returning the same response.. can we consolidate them into a single block?

Reply:
Done